src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java

```diff
@@ -72,6 +78,11 @@
      */
     private static final VoxelShape SHAPE = VoxelShapes.create(0.1, 0.1, 0.1, 0.9, 0.9, 0.9);
 
+    /**
+     * Whether this hut is yet to be researched in the current colony.
+     */
+    protected boolean needsResearch = false;
```

Comment:
I will look this up later, but isn't `addInformation` client side only?
In that case its perfectly fine if its the case for every instance of the block, because its only whether or not the tooltip gets rendered for this player.

Reply:
I looked it up, and in `BlockItem` `addInformation` is `@OnlyIn(Dist.CLIENT)`, so i think it should be fine on servers too.
I will test whether it actually works on a dedicated server, but i don't know why it wouldn't.

Reply:
Finally figured out why it might not work xD
Its because the colony check only works server side, but is nbt really the only solution?
Couldn't the client store it somewhere, and it wouldn't be that much of a performance hit.
The client needs to know the researches anywhere anyways, can't i just use that for the check then?

Reply:
The research effects are synched to the client afaik

Reply:
I know that addInformation is Client only, that's not what I am referring to. It's the other stuff that isn't client side only that may cause problems, like the needsResearch variable on the server (which is not purely Client based). It will be reflected through all the other blockstates. Sorry if I'm not making sense, but this problem is probably not very obvious with only one player.

Reply:
> this problem is probably not very obvious with only one player.

It is. I just tested it, and the tooltips just aren't showing up at all on a dedicated server.
> The research effects are synched to the client afaik

Thx, i will look for them.

Reply:
The only thing not working with the new commit, is if you leave a colony into wilderness, it keeps all the tooltips, even after going into another world.